Final changelog and upgrade guide (#3135)
plone/volto: CHANGELOG.md

### Breaking

- Upgrade `react-cookie` to the latest version. @sneridagh @robgietema

>   See https://docs.voltocms.com/upgrade-guide/ for more information.

Reviewer: Do we want to use that URL or the temporary one?

https://6.dev-docs.plone.org/volto/upgrade-guide/index.html

> - Language Switcher no longer takes care of the change of the language on the Redux Store. This responsibility has been unified in the Api Redux middleware @sneridagh

Reviewer: ```suggestion
- Language Switcher no longer takes care of the change of the language on the Redux Store. This responsibility has been unified in the API Redux middleware @sneridagh
```

- Markup change in `LinkView` component.
- Rename `core-sandbox` to `coresandbox` for sake of consistency @sneridagh
- Extend the original intent and rename `RAZZLE_TESTING_ADDONS` to `ADDONS`. @sneridagh

>   See https://docs.voltocms.com/upgrade-guide/ for more information.

Reviewer: Use this URL or https://6.dev-docs.plone.org/volto/upgrade-guide/index.html ?

> - Lazyload draftjs library. See the upgrade guide on how that impacts you, in case you have extended the rich text editor configuration @tiberiuichim @kreafox

Reviewer: ```suggestion
- Lazy load Draft.js library. See the upgrade guide on how that impacts you, in case you have extended the rich text editor configuration @tiberiuichim @kreafox
```

>   See https://docs.voltocms.com/upgrade-guide/ for more information.

Reviewer: Use this URL or https://6.dev-docs.plone.org/volto/upgrade-guide/index.html ?

- Deprecating `lang` cookie in favor of Plone official one `I18N_LANGUAGE` @sneridagh

### Feature

> - Add `cookiesExpire` value to config to control the cookie expiration @giuliaghisini

Reviewer: ```suggestion
- Add `cookiesExpire` value to configure the cookie expiration @giuliaghisini
```

> - DatetimeWidget 'noPastDates' option: Take widgetOptions?.pattern_options?.noPastDates of backend schema into account. @ksuess

Reviewer: Are there supposed to be "?" in this change log entry?

- Add a new type of filter facet for the Search block. Heavily refactor some searchblock internals. @tiberiuichim
- Add date range facet to the search block @robgietema
- Introduce the new `BUILD_DIR` runtime environment variable to direct the build to run in a specific location, different than `build` folder. @sneridagh
- Handle redirect permanent calls from the backend in the frontend (e.g. when changing the short name) @robgietema
- Added id widget to manage short name @robgietema

> - Refactor language synchronizer. Remove it from the React tree, integrate it into the Api Redux middleware @sneridagh

Reviewer: ```suggestion
- Refactor language synchronizer. Remove it from the React tree, integrate it into the API Redux middleware @sneridagh
```

Reviewer: Also is this a duplicate entry from line 9?

- Add blocks rendering in Event and NewsItem views (rel plone.volto#32) @nzambello @ksuess
- Add internal volto ids to invalid ids @robgietema

> - Complete basque translation @erral

Reviewer: ```suggestion
- Complete Basque translation @erral
```

> - Complete spanish translation @erral

Reviewer: ```suggestion
- Complete Spanish translation @erral
```

- Sort the choices in Facets in the search block @iFlameing

### Bugfix

- Fix the `null` error in SelectAutoComplete Widget @iFlameing

> - Prevent the MultilingualRedirector to force 4 content load when switching the language @reebalazs

Reviewer: Why is there a "4" in here?

```suggestion
- Prevent the MultilingualRedirector to force content load when switching the language @reebalazs
```

- Fix the upload image in contents view @iFlameing
- add "view" id to contact-form container for main content skiplink @ThomasKindermann
- Fix loading indicator positioning on Login form submit @sneridagh
- Fix redirect bug with URLs containing querystrings @robgietema
- Fixed id widget translations @robgietema
- Contents Rename Modal, use `id` Widget type @sneridagh
- Fix overflow of very long file name in `FileWidget` @sneridagh
- Fix overflowing issue in the toolbar @kreafox

### Internal

- Change prop `name` -> `componentName` in component `Component` @sneridagh
- Add new RawMaterial Volto websites in production @nzambello
- House cleanup, remove some unused files in the root @sneridagh
- Move Webpack related files to `webpack-plugins` folder @sneridagh
- Remove unused Dockerfiles @sneridagh
- Update Docker compose to latest images and best practices @sneridagh
- Improve flaky test in coresandbox search Cypress tests @sneridagh
- Better implementation of the add-on load coming from the environment variable `ADDONS` @sneridagh

> - Turn `lazyLibraries` action into a thunk. Added a conditional if the library is loaded or in process to be loaded, do not try to load it again. This fixes the lag on load `draftjs` when having a lot of draftjs blocks. @sneridagh

Reviewer: "thunk"? perhaps "chunk"?

Author: it's fine, this artifact is called like this.
